src/connectors/meraki_devices.py

```diff
@@ -0,0 +1,194 @@
+"""Meraki Devices
+Collect Meraki Device information using an API Token
+"""
+
+from runners.helpers import log
+from runners.helpers import db
+from runners.helpers.dbconfig import ROLE as SA_ROLE
+
+from datetime import datetime
+
+import snowflake
+import requests
+from urllib.error import HTTPError
+from .utils import yaml_dump
+
+
+
+PAGE_SIZE = 5
+
+
+CONNECTION_OPTIONS = [
+    {
+        'name': 'api_token',
+        'title': "Meraki API Token",
+        'prompt': "Your Meraki API Token",
+        'type': 'str',
+        'secret': True,
+        'required': True,
+    },
+    {
+        'name': 'network_id_whitelist',
+        'title': "Meraki Network Ids Whitelist",
+        'prompt': "Whitelist of Network Ids",
+        'type': 'list',
+        'secret': False,
+        'required': True,
+    },
+]
+
+LANDING_TABLE_COLUMNS_CLIENT = [
+    ('INSERT_ID', 'NUMBER IDENTITY START 1 INCREMENT 1'),
+    ('SNAPSHOT_AT', 'TIMESTAMP_LTZ(9)'),
+    ('RAW', 'VARIANT'),
+    ('ID', 'VARCHAR(256)'),
+    ('MAC', 'VARCHAR(256)'),
+    ('DESCRIPTION', 'VARCHAR(256)'),
+    ('MDNS_NAME', 'VARCHAR(256)'),
+    ('DHCP_HOSTNAME', 'VARCHAR(256)'),
+    ('IP', 'VARCHAR(256)'),
+    ('SWITCHPORT', 'VARCHAR(256)'),
+    ('VLAN', 'INT'), 
+    ('USAGE_SENT', 'INT'),
+    ('USAGE_RECV', 'INT'),
+    ('SERIAL', 'VARCHAR(256)'),
+]
+
+LANDING_TABLE_COLUMNS_DEVICE = [
+    ('INSERT_ID', 'NUMBER IDENTITY START 1 INCREMENT 1'),
+    ('SNAPSHOT_AT', 'TIMESTAMP_LTZ(9)'),
+    ('RAW', 'VARIANT'),
+    ('SERIAL', 'VARCHAR(256)'),
+    ('ADDRESS', 'VARCHAR(256)'),
+    ('NAME', 'VARCHAR(256)'),
+    ('NETWORK_ID', 'VARCHAR(256)'),
+    ('MODEL', 'VARCHAR(256)'),
+    ('MAC', 'VARCHAR(256)'),
+    ('LAN_IP', 'VARCHAR(256)'),
+    ('WAN_1_IP', 'VARCHAR(256)'),
+    ('WAN_2_IP', 'VARCHAR(256)'),
+    ('TAGS', 'VARCHAR(256)'),
+    ('LNG', 'FLOAT'),
+    ('LAT', 'FLOAT'),
+]
+
+
+def get_data(url: str, token: str, params: dict = {}) -> dict:
+    headers: dict = {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "X-Cisco-Meraki-API-Key": f"{token}",
+    }
+    try:
+        log.debug(f"Preparing GET: url={url} with params={params}")
+        req = requests.get(url, params=params, headers=headers)
+        req.raise_for_status()
+    except HTTPError as http_err:
+        log.error(f"Error GET: url={url}")
+        log.error(f"HTTP error occurred: {http_err}")
+        raise
+    log.debug(req.status_code)
+    return req.json()
+
+
+def connect(connection_name, options):
+    landing_table_client = f'data.meraki_devices_{connection_name}_connection_client'
+    landing_table_device = f'data.meraki_devices_{connection_name}_connection_device'
+    options['network_id_whitelist'] = options.get('network_id_whitelist', '').split(',')
```

Comment:
Would the special UI element parse the user input as a list? When we tested this particular input, we got it as a string, like `'id1, id2, id3'`.

Reply:
https://github.com/snowflakedb/SnowAlert/pull/306

Reply:
Perfect, thanks.

Err, had a bit of a think-o. The "return" type I was thinking of was of the expression

    options.get('network_id_whitelist', '').split(',')

but since it splits the empty string, it's always a list.